site/docs/contributing.md

```diff
@@ -0,0 +1,267 @@
+---
+title: Contributing to promptfoo
+sidebar_label: Contributing
+---
+
+We welcome contributions from the community to help make promptfoo better. This guide will help you get started. If you have any questions, please reach out to us on [Discord](https://discord.gg/gHPS9jjfbs) or through a [GitHub issue](https://github.com/promptfoo/promptfoo/issues/new).
+
+## Project Overview
+
+promptfoo is an MIT licensed tool for testing and evaluating LLM apps.
+
+We particularly welcome contributions in the following areas:
+
+- Bug fixes
+- Documentation updates, including examples and guides
+- Updates to providers including new models, new capabilities (tool use, function calling, JSON mode, file uploads, etc.)
+- Features that improve the user experience of promptfoo, especially relating to RAGs, Agents, and synthetic data generation.
+
+## Getting Started
+
+1. Fork the repository on GitHub.
+2. Clone your fork locally:
+
+   ```sh
+   git clone https://github.com/[your-username]/promptfoo.git
+   cd promptfoo
+   ```
+
+3. Set up your development environment:
+
+   ```sh
+   # We recommend using the Node.js version specified in the .nvmrc file (ensure node >= 18)
+   nvm use
+   npm install
+   ```
+
+4. Run the tests to make sure everything is working:
```

Comment:
Consider replacing steps 4 and 5 w/ instructions on running the API/Web UI/CLI with hotreloading and moving test and build into Development Workflow w/ links to the relevant (subsequent) sections.

Reply:
This is a good idea. Thank you for the suggestion. I am going to keep it as is for now because we only have tests on the cli. I will start to refactor it in this way as we build out tests on the frontend / docs / etc.